crates/router/src/types/api/webhooks.rs

```diff
@@ -73,10 +76,71 @@ pub trait IncomingWebhook: ConnectorCommon + Sync {
 
     async fn get_webhook_source_verification_merchant_secret(
         &self,
-        _db: &dyn StorageInterface,
-        _merchant_id: &str,
+        db: &dyn StorageInterface,
+        merchant_id: &str,
+        connector_label: &str,
     ) -> CustomResult<Vec<u8>, errors::ConnectorError> {
-        Ok(Vec::new())
+        let debug_suffix = format!(
+            "For merchant_id: {}, and connector_label: {}",
+            merchant_id, connector_label
+        );
+        let key_store = db
+            .get_merchant_key_store_by_merchant_id(
+                merchant_id,
+                &db.get_master_key().to_vec().into(),
+            )
+            .await
+            .change_context(errors::ConnectorError::WebhookSourceVerificationFailed)
+            .attach_printable_lazy(|| {
+                crate::logger::warn!("Fetch  from MCA table failed {}", debug_suffix);
+                format!(
+                    "Fetch merchant_webhook_secret from MCA table failed {}",
+                    debug_suffix
+                )
```

Comment:
both log and attaching printable is not necessary, can you please remove the log statement here, and in other places too?